lib/restbase.js

```diff
@@ -32,12 +32,14 @@ function RESTBase (options, req) {
         var par = this._parent = options;
         this.log = par.log;
         this.metrics = par.metrics;
+        this.reqId = req && req.headers && req.headers['x-request-id'] || req && req.reqId || par.reqId || rbUtil.generateRequestId();
```

Comment:
`par.reqId` would be the most common case, so it's probably fastest to check that first. Could you also break this long line?